pkg/devfile/generator/utils.go

```diff
@@ -619,11 +623,71 @@ func getAllContainers(devfileObj parser.DevfileObj, options common.DevfileOption
 				return nil, err
 			}
 		}
-		containers = append(containers, *container)
+		modifiedContainer, err := containerOverridesHandler(comp, container)
+		if err != nil {
+			return nil, err
+		}
+		if modifiedContainer != nil {
+			containers = append(containers, *modifiedContainer)
+		} else {
+			containers = append(containers, *container)
+		}
 	}
 	return containers, nil
 }
 
+// containerOverridesHandler modifies the container component to include any container-overrides values in the devfile.
+// It does so by merging the devfile JSON with the JSON string representation of container-overrides field
+func containerOverridesHandler(comp v1.Component, container *corev1.Container) (*corev1.Container, error) {
```

Comment:
I find it a bad idea to pass a pointer and return a pointer. But I'm struggling to do otherwise because of the way `runtime.DefaultUnstructuredConverter.ToUnstructured` works.

I'm open to ideas/suggestions.